### ISSUE TITLE
Error: [string "?"]:696: error opening file.

### ISSUE THREAD

**Issue by Ratismal** [Rinnegatamante/CHMM#16]
Hi, I'm sorry to bother you again.

When I try to use shuffle mode, I get the same error as mentioned [on this post](https://gbatemp.net/threads/release-chmm2-custom-home-menu-manager-2.397543/page-92#post-6511535). I have tried searching around the thread for solution, deleting the `extdata`, deleting the `settings.cfg` file, and even renaming all of my folders, but to no avail. I should note that all of my themes work individually, and this error only appears when I try to apply a shuffle set. I'm using the .3dsx version on v2.8.

For reference, this is the layout of my `/3ds/CHMM2/Themes` folder
![themes](https://i.gyazo.com/074e8a2fabe3e41a5d11d1d8975a76e2.png)

Any help with this would be greatly appreciated. Thank you for your time.

**Comment by Rinnegatamante** [comment]
https://github.com/Rinnegatamante/CHMM2/blob/master/source/themes.lua#L242

This is where your error happens. Another time, seems like extdata cache is not well generated.
Are you on fw > 9.2?
If so, try to set shuffle themes with Nintendo themes manager and then try to use CHMM2.

**Comment by Ratismal** [comment]
That worked the first time I tried it, using two different themes. However, when I tried to assign more than that, it would fail with the following error:

```
Error: [string "?"]:728: error opening file.
```

Repeated attempts would create more and more errors, ranging from not being able to install single themes to not being able to access the extdata. Since then, I've tried to follow the steps again, including deleting the theme extdata, but am unable to even make a shuffle set of the original two themes. I've yet to try deleting the homemenu extdata, as it messed with all my icons, but I'll try doing that along with the theme extdata if I can't get this to work.

EDIT: After deleting both extdatas, I am able to set a shuffled set with two themes again. Any more, and I get the error at line 728.

**Comment by Rinnegatamante** [comment]
Looks pretty strange, are you using 3DSX version right? It seems like on latest fw CHMM2 is having a lot of similar troubles in the 3DSX format. Try to install a 10 themes shuffle with nintendo themes manager and then try with CHMM2.

**Comment by Ratismal** [comment]
Yes, 3dsx version on the latest firmware. I'm going to give step by step descriptions of what I'm doing and what the result was, in hopes that it will help.
1. Deleted the extdata folders (US version, so 0000008f for the home menu and 000002cd for the theme)
2. Generate home menu data (just by turning on ds)
3. Go into theme menu, and generate theme data
4. Go into `Shuffle Favorites` and assign ten nintendo-originated themes. Theme-management information is updated. No problems thus far.
5. Go to the homebrew channel via web browser ([3dsbrowserhax](http://yls8.mtheall.com/3dsbrowserhax_auto.php))
6. Open CHMM2, or BLEND3R as it's called in the homebrew launcher. Haven't bothered to fix that.
7. Press 'X' to open the shuffle menu.
8. Assign themes to the first 6 slots.
9. Press 'X' to save and install themeset. At this point, the `Error: [string "?"]:728: error opening file.` message is encountered.
10. Press 'A' to restart app.
11. Press 'A' again to install a single theme. It installs with no issue.
12. Repeat steps 7-9, but with 2 themes instead of 6.
13. Installs successfully.
14. Exit to home menu to test if the applied themes work. Theme one is loaded.
15. Closed 3ds for a while to see if second theme would load. It does.

So, shufflesets are working perfectly fine for me, provided I don't use more than two themes. I hope this helps you!

**Comment by Rinnegatamante** [comment]
Pretty random stuffs T.T I hate Nintendo. I'll try to add more "security" code for extdata themes files in next release.

**Comment by Ratismal** [comment]
I hope you get it working! For now, I can make do with two themes. Thanks for all your help!

**Comment by oliverdom** [comment]
[string ''?'']: 625: error opening file.
Could you tell me how to fix it?

**Comment by eryns-works** [comment]
I got mine working. I'm using Luma3DS, on sysNAND.

I deleted the config file in SDHC:\Themes\, re-inserted the SD card to my 3DS. In the official Nintendo Themes Manager, I created a shuffle of four default themes. Then, in CHMM2, I created a shuffle of four custom themes. After setting my themes in CHMM2 and pressing X to enable shuffling, the app seemed to freeze for a little while, but it did unfreeze eventually. When it did, I pressed start, then clicked Exit CHMM2. My 3DS restarted and now all four themes are shuffling properly, no errors.
